src/Infrastructure/FeatureFlagService.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Infrastructure;
+
+use Symfony\Component\HttpFoundation\Request;
+
+class FeatureFlagService
+{
+    private array $env;
+
+    public function __construct(array $env = null)
+    {
+        $this->env = $env ?? $_ENV;
```

Comment:
Ce serait préférable de passer par l'injection de dépendance proposer par symfony et ne pas utiliser la variable globale $_ENV.

1. Définir dans la configuration bind `featureNewTrucEnabled: %env(bool:FEATURE_NEW_TRUC_ENABLED%)`.

2. Injecter la propriété `$featureNewTrucEnabled` partout où on en a besoin.

Qu'en penses-tu ?

Reply:
Je t'avoue que je n'en verrais pas l'utilité ; ça serait plus de verbiage de config pour faire la même chose non ? Et devoir définir un nouveau bind à chaque fois qu'on veut définir et utiliser un nouveau feature flag...

Reply:
Pour tester il faudrait configurer un feature flag "de test" dans `services.yml` ?

Reply:
Je vois que tout serait plus verbeux

Soit on ferait une méthode par feature flag (bof)

Soit dans `isFeatureEnabled($featureName)` on feait un `if ($featureName === 'SOME_FEATURE') return $this->isSomeFeatureEnabled`, donc un if explicite par feature flag

Je trouve le côté dynamique de l'implé actuelle sympa pour implémenter "l'infrastructure" une bonne fois pour toute

Reply:
À noter aussi que je fais de l'injection de dépendance car je traite `$_ENV` comme une valeur par défaut qu'on peut remplacer à des fins de testabilité...

Reply:
Je suis d'accord avec l'utilisation d'un service centralisé, moins sur l'utilisation de la variable `$_ENV` qui n'est pas très symfony compliant.
Il existe plein d'exemples d'implémentation sur la toile dont on pourrait s'inspirer ? Par ex (https://github.com/novaway/NovawayFeatureFlagBundle, https://smaine-milianni.medium.com/feature-flag-and-strategy-pattern-with-the-symfony-framework-7863ecc9556a).
Ou alors, on pourrait très bien prendre le parti d'installer un bundle, qui de toute façon n'aura pas vocation à durer dans le temps.
Je chipote un peu, mais je n'aime pas trop voir l'utilisation des variables globales au sein du projet. Plus on s'en passe, mieux c'est.

Reply:
OK ça marche. J'ai créé un paramètre `features` où on pourra mettre par ex :

```yaml
parameters:
  features:
    road_type: '%env(APP_FEATURE_ROAD_TYPE_ENABLED)%'
    loc_inversion: '%env(APP_FEATURE_LOC_INVERSION_ENABLED)%'
```

cc @Lealefoulon